lib/matplotlib/backends/backend_pgf.py

```diff
@@ -600,6 +600,21 @@ def _print_pgf_path(self, gc, path, transform, rgbFace=None):
                         r"{\pgfqpoint{%fin}{%fin}}"
                         % coords)
 
+        # apply pgf decorators
+        sketch_params = gc.get_sketch_params() if gc else None
+        if sketch_params is not None:
+            # Only "length" directly maps to "segment length" in PGF's API
+            # The others are combined in "amplitude" -> Use "randomness" as
+            # PRNG seed to allow the user to force the same shape on multiple
+            # sketched lines
```

Comment:
I'm confused by this comment, as there's only one thing in `amplitude`.

Reply:
Well, matplotlib: https://matplotlib.org/stable/api/_as_gen/matplotlib.artist.Artist.set_sketch_params.html and pgf: https://mirror.physik.tu-berlin.de/pub/CTAN/graphics/pgf/base/doc/pgfmanual.pdf#subsubsection.50.3.1 (page 640, "Decoration random steps") do not exactly fit together.

PGF only has "segment length" and "amplitude" as arguments (randomness coming from a built-in RNG).
"amplitude" changes both the length of the wiggle along the line AND the wiggle perpendicular to the source line.

From my understanding, the matplotlib docs say that it only varies the "(segment) length" by a given randomized factor of "randomness" (without randomizing the wiggle perpendicular to the source line).

I updated the comment, should be more clear now.

Reply:
The `length` defines the wavelength of the sine wave used for perturbation. The `randomness` determines how quickly each segment goes through the sine wave. So I guess they are kind of two arguments to the same thing.

See
https://github.com/matplotlib/matplotlib/blob/246a489ce628a8f0d9cd6f566d5ce09a4f2b5454/src/path_converters.h#L978-L994

Reply:
Yes, and the real amplitude (as in a 1D sine wave) is not randomized by `randomness` (which could be considered as a separate bug, imho).

PGF uses a completely different approach:
The line is split into segments of length `segment length` which walk towards the destination point.
The end coordinate of each line segment (aka step) is randomized by at most ± `amplitude`, independently in both X- and Y direction:

https://github.com/pgf-tikz/pgf/blob/247f1f0445a9e996560c9f5a9b3fb03a4184e6ec/tex/generic/pgf/libraries/decorations/pgflibrarydecorations.pathmorphing.code.tex#L86-L101

I didn't want to adapt matplotlib to pgf's behaviour more than necessary, so I just settled on this comment...